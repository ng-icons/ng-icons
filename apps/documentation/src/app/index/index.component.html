<div
  appFadeInContainer
  class="flex flex-col items-center justify-center h-screen px-4 mt-6 md:mt-0"
>
  <img
    appFadeIn
    class="w-32 mb-6"
    src="assets/angular.svg"
    alt="Angular Logo"
  />

  <h2
    appFadeIn
    class="heading mb-5 text-base md:text-xl font-extrabold uppercase"
  >
    Angular Icons
  </h2>

  <h1
    appFadeIn
    class="slogan font-extrabold text-3xl md:text-5xl text-center mb-6 md:leading-[60px]"
  >
    The all-in-one icon<br />
    library for Angular
  </h1>

  <p appFadeIn class="text-sm md:text-base text-[#F5EAEA] text-center mb-16">
    Providing over 48,000 icons for use in your applications
  </p>

  <div appFadeIn class="flex gap-6 flex-col md:flex-row">
    <button
      routerLink="/getting-started"
      class="btn btn-white h-12 px-6 flex gap-x-2 items-center font-medium text-sm transition-shadow ease-out duration-200"
    >
      <span class="mb-0.5">Get Started</span>

      <ng-icon name="heroChevronRightMini" class="text-xl"></ng-icon>
    </button>

    <button
      routerLink="/browse-icons"
      class="btn btn-red h-12 px-6 flex gap-x-2 items-center font-medium text-sm transition-shadow ease-out duration-200"
    >
      <span class="mb-0.5">Browse Icons</span>

      <ng-icon name="heroChevronRightMini" class="text-xl"></ng-icon>
    </button>
  </div>
</div>
