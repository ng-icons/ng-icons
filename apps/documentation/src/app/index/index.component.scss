.heading {
  line-height: 25px;
  letter-spacing: 0.08em;
  background: linear-gradient(360deg, #ffafbd 0%, #ffc3a0 82%);
  -webkit-background-clip: text;
  -webkit-text-fill-color: transparent;
  background-clip: text;
  text-fill-color: transparent;
}

.slogan {
  line-height: 60px;
  background: linear-gradient(
    180deg,
    #ffffff 0%,
    rgba(227, 153, 153, 0.96) 100%
  );
  -webkit-background-clip: text;
  -webkit-text-fill-color: transparent;
  background-clip: text;
  text-fill-color: transparent;
}

.btn {
  border-radius: 32px;
  box-shadow: 0 0 0 rgba(0, 0, 0, 0);

  &.btn-white {
    background: linear-gradient(90deg, #e0eafc 0%, #cfdef3 100%);
    border-top: 1px solid #ffffff;
    color: #1a1e2a;

    &:hover {
      box-shadow: 0px 1px 40px rgba(224, 234, 252, 0.4);
    }
  }

  &.btn-red {
    border-top: 1px solid rgba(238, 238, 238, 0.3);
    background: linear-gradient(90deg, #e53935 0%, #e35d5b 100%);
    color: #fff;

    &:hover {
      box-shadow: 0px 1px 40px rgba(216, 24, 47, 0.5);
    }
  }
}
