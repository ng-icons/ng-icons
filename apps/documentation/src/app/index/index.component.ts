import { Component } from '@angular/core';
import { RouterLink } from '@angular/router';
import { NgIconComponent, provideIcons } from '@ng-icons/core';
import { heroChevronRightMini } from '@ng-icons/heroicons/mini';
import { FadeInContainerDirective } from '../directives/fade-in/fade-in-container.directive';
import { FadeInDirective } from '../directives/fade-in/fade-in.directive';

@Component({
  selector: 'app-index',
  templateUrl: './index.component.html',
  styleUrls: ['./index.component.scss'],
  standalone: true,
  imports: [
    NgIconComponent,
    RouterLink,
    FadeInContainerDirective,
    FadeInDirective,
  ],
  providers: [provideIcons({ heroChevronRightMini })],
})
export class IndexComponent {}
