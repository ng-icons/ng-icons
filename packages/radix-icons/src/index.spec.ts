import * as icons from './index';

describe('Radix Icons', () => {
  it('should ensure the export names have not changed unexpectedly', () => {
    expect(Object.keys(icons)).toMatchSnapshot();
  })
});
