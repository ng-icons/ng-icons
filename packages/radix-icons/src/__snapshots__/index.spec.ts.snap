// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Radix Icons should ensure the export names have not changed unexpectedly 1`] = `
Array [
  "radixActivityLog",
  "radixAlignBaseline",
  "radixAlignBottom",
  "radixAlignCenterHorizontally",
  "radixAlignCenterVertically",
  "radixAlignCenter",
  "radixAlignEnd",
  "radixAlignHorizontalCenters",
  "radixAlignLeft",
  "radixAlignRight",
  "radixAlignStart",
  "radixAlignStretch",
  "radixAlignTop",
  "radixAlignVerticalCenters",
  "radixAllSides",
  "radixAngle",
  "radixArchive",
  "radixArrowBottomLeft",
  "radixArrowBottomRight",
  "radixArrowDown",
  "radixArrowLeft",
  "radixArrowRight",
  "radixArrowTopLeft",
  "radixArrowTopRight",
  "radixArrowUp",
  "radixAspectRatio",
  "radixAvatar",
  "radixBackpack",
  "radixBadge",
  "radixBarChart",
  "radixBell",
  "radixBlendingMode",
  "radixBookmark",
  "radixBorderAll",
  "radixBorderBottom",
  "radixBorderDashed",
  "radixBorderDotted",
  "radixBorderLeft",
  "radixBorderNone",
  "radixBorderRight",
  "radixBorderSolid",
  "radixBorderSplit",
  "radixBorderStyle",
  "radixBorderTop",
  "radixBoxModel",
  "radixBox",
  "radixButton",
  "radixCalendar",
  "radixCamera",
  "radixCardStackMinus",
  "radixCardStackPlus",
  "radixCardStack",
  "radixCaretDown",
  "radixCaretLeft",
  "radixCaretRight",
  "radixCaretSort",
  "radixCaretUp",
  "radixChatBubble",
  "radixCheckCircled",
  "radixCheck",
  "radixCheckbox",
  "radixChevronDown",
  "radixChevronLeft",
  "radixChevronRight",
  "radixChevronUp",
  "radixCircleBackslash",
  "radixCircle",
  "radixClipboardCopy",
  "radixClipboard",
  "radixClock",
  "radixCode",
  "radixCodesandboxLogo",
  "radixColumnSpacing",
  "radixColumns",
  "radixCommit",
  "radixComponent1",
  "radixComponent2",
  "radixComponentBoolean",
  "radixComponentInstance",
  "radixComponentNone",
  "radixComponentPlaceholder",
  "radixContainer",
  "radixCookie",
  "radixCopy",
  "radixCornerBottomLeft",
  "radixCornerBottomRight",
  "radixCornerTopLeft",
  "radixCornerTopRight",
  "radixCorners",
  "radixCountdownTimer",
  "radixCounterClockwiseClock",
  "radixCrop",
  "radixCross1",
  "radixCross2",
  "radixCrossCircled",
  "radixCrosshair1",
  "radixCrosshair2",
  "radixCrumpledPaper",
  "radixCube",
  "radixCursorArrow",
  "radixCursorText",
  "radixDash",
  "radixDashboard",
  "radixDimensions",
  "radixDisc",
  "radixDividerHorizontal",
  "radixDividerVertical",
  "radixDotFilled",
  "radixDotSolid",
  "radixDot",
  "radixDotsHorizontal",
  "radixDotsVertical",
  "radixDoubleArrowDown",
  "radixDoubleArrowLeft",
  "radixDoubleArrowRight",
  "radixDoubleArrowUp",
  "radixDownload",
  "radixDragHandleDots1",
  "radixDragHandleDots2",
  "radixDragHandleHorizontal",
  "radixDragHandleVertical",
  "radixDrawingPinFilled",
  "radixDrawingPinSolid",
  "radixDrawingPin",
  "radixDropdownMenu",
  "radixEnterFullScreen",
  "radixEnter",
  "radixEnvelopeClosed",
  "radixEnvelopeOpen",
  "radixExclamationTriangle",
  "radixExitFullScreen",
  "radixExit",
  "radixExternalLink",
  "radixEyeClosed",
  "radixEyeNone",
  "radixEyeOpen",
  "radixFace",
  "radixFigmaLogo",
  "radixFileMinus",
  "radixFilePlus",
  "radixFileText",
  "radixFile",
  "radixFontBold",
  "radixFontFamily",
  "radixFontItalic",
  "radixFontRoman",
  "radixFontSize",
  "radixFontStyle",
  "radixFrame",
  "radixFramerLogo",
  "radixGear",
  "radixGithubLogo",
  "radixGlobe",
  "radixGrid",
  "radixGroup",
  "radixHalf1",
  "radixHalf2",
  "radixHamburgerMenu",
  "radixHand",
  "radixHeading",
  "radixHeight",
  "radixHome",
  "radixIconjarLogo",
  "radixIdCard",
  "radixImage",
  "radixInfoCircled",
  "radixInnerShadow",
  "radixInput",
  "radixJustifyCenter",
  "radixJustifyEnd",
  "radixJustifyStart",
  "radixJustifyStretch",
  "radixLapTimer",
  "radixLayers",
  "radixLayout",
  "radixLetterCaseCapitalize",
  "radixLetterCaseLowercase",
  "radixLetterCaseToggle",
  "radixLetterCaseUppercase",
  "radixLetterSpacing",
  "radixLightningBolt",
  "radixLineHeight",
  "radixLink1",
  "radixLink2",
  "radixLinkBreak1",
  "radixLinkBreak2",
  "radixLinkNone1",
  "radixLinkNone2",
  "radixListBullet",
  "radixLockClosed",
  "radixLockOpen1",
  "radixLockOpen2",
  "radixLoop",
  "radixMagicWand",
  "radixMagnifyingGlass",
  "radixMargin",
  "radixMaskOff",
  "radixMaskOn",
  "radixMinusCircled",
  "radixMinus",
  "radixMix",
  "radixMixerHorizontal",
  "radixMixerVertical",
  "radixModulzLogo",
  "radixMoon",
  "radixMove",
  "radixNotionLogo",
  "radixOpacity",
  "radixOuterShadow",
  "radixOverline",
  "radixPadding",
  "radixPaperPlane",
  "radixPause",
  "radixPencil1",
  "radixPencil2",
  "radixPerson",
  "radixPieChart",
  "radixPilcrow",
  "radixPinBottom",
  "radixPinLeft",
  "radixPinRight",
  "radixPinTop",
  "radixPlay",
  "radixPlusCircled",
  "radixPlus",
  "radixQuestionMarkCircled",
  "radixQuestionMark",
  "radixQuote",
  "radixRadiobutton",
  "radixReader",
  "radixReload",
  "radixReset",
  "radixResume",
  "radixRocket",
  "radixRotateCounterClockwise",
  "radixRowSpacing",
  "radixRows",
  "radixRulerHorizontal",
  "radixRulerSquare",
  "radixSection",
  "radixSewingPinFilled",
  "radixSewingPinSolid",
  "radixSewingPin",
  "radixShadowInner",
  "radixShadowNone",
  "radixShadowOuter",
  "radixShadow",
  "radixShare1",
  "radixShare2",
  "radixShuffle",
  "radixSize",
  "radixSketchLogo",
  "radixSlash",
  "radixSlider",
  "radixSpaceBetweenHorizontally",
  "radixSpaceBetweenVertically",
  "radixSpaceEvenlyHorizontally",
  "radixSpaceEvenlyVertically",
  "radixSpeakerLoud",
  "radixSpeakerModerate",
  "radixSpeakerOff",
  "radixSpeakerQuiet",
  "radixSquare",
  "radixStack",
  "radixStarFilled",
  "radixStar",
  "radixStitchesLogo",
  "radixStop",
  "radixStopwatch",
  "radixStretchHorizontally",
  "radixStretchVertically",
  "radixStrikethrough",
  "radixSun",
  "radixSwitch",
  "radixSymbol",
  "radixTable",
  "radixTarget",
  "radixTextAlignCenter",
  "radixTextAlignJustify",
  "radixTextAlignLeft",
  "radixTextAlignRight",
  "radixTextNone",
  "radixText",
  "radixThickArrowDown",
  "radixThickArrowLeft",
  "radixThickArrowRight",
  "radixThickArrowUp",
  "radixTimer",
  "radixTokens",
  "radixTrackNext",
  "radixTrackPrevious",
  "radixTransparencyGrid",
  "radixTrash",
  "radixTriangleDown",
  "radixTriangleLeft",
  "radixTriangleRight",
  "radixTriangleUp",
  "radixTwitterLogo",
  "radixUnderline",
  "radixUpdate",
  "radixUpload",
  "radixValueNone",
  "radixValue",
  "radixVercelLogo",
  "radixVideo",
  "radixViewGrid",
  "radixViewHorizontal",
  "radixViewNone",
  "radixViewVertical",
  "radixWidth",
  "radixZoomIn",
  "radixZoomOut",
]
`;
